### PR TITLE
Add type definitions

### LINKED ISSUE #91: TypeScript support
Will there be added support for using this component with react typeScript out of the box, in the near future?

Error message when trying to incorporate this in a typescript project:
JSX element type 'SuperSelectField' does not have any construct or call signatures

### PR DESCRIPTION
**Pull request by FDiskas**
Closes #91

### CODE REVIEW

**Comment by Sharlaan** [comment]
Can you update or remake this PR or rebase, this time based on new master branch please ?

File: lib/SuperSelectField.d.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+declare module 'SuperSelectField' {
+    import {Component} from 'react';
+
+    export interface SelectFieldProps {
+        anchorOrigin?: any;
```
Comment (FDiskas) [inline lib/SuperSelectField.d.ts]: Why everywhere types are any?
https://github.com/KnisterPeter/react-to-typescript-definitions/issues/439

Comment (FDiskas) [reply lib/SuperSelectField.d.ts]: and this file should be in ignore to :)

Comment (Sharlaan) [reply lib/SuperSelectField.d.ts]: Speaking to yourself ? ^^

**Comment by FDiskas** [comment]
Useless codecov comment. Anybody knows how to turn off commenting on pull requests?

**Comment by Sharlaan** [comment]
One question:
Have you tried installing and using both the es5 and module versions of material-ui-superselectifled, on a dummy page featuring react/react-dom/material-ui ?

I'm asking this following your modification of package.json properties "main" and "module".

**Comment by FDiskas** [comment]
I tried to `npm run build` and that index.js was missing :poodle:
Logic is simple. 
* User installs `material-ui-superselectfield`
* Users imports `import * as Select from 'material-ui-superselectfield'`
* Webpack (or whatever is) will look in node_modules the directory `material-ui-superselectfield`
* Opens modules package.json file and looks for "main" section, because user was not specified the file in import statment

**Comment by Sharlaan** [comment]
If you read NWB documentation as suggested few days ago, you would have learn that :
lib/ contains the es5 version, transpiled
es/ contains the module-compatible version, by extension untranspiled since module import is last EcmaScript feature implemented into nowadays browsers
so this means :
1. you MUST include these folders in npm publish
2. index.js allows the browser JS engine to get all needed files (reminder they are splitted !) by default
So if you rename this file browser won't import properly

There is a third build, it were the original form of my project before this transfer to NWB: the UMD.
In this particular build case, yea all component files are concatenated into one.
We didnot include this one, es5 and module versions should be enough for most projects.

**Comment by FDiskas** [comment]
You should check the results after pre released
https://unpkg.com/material-ui-superselectfield/ and check the "main" section in package.json

**Comment by Sharlaan** [comment]
What your link is pointing at, is a "UMD" build.
As explained above, we decided to ditch this in favor of es5 and es6+ module versions...

**Comment by FDiskas** [comment]
Yes link is pointing to the latest npm module version. Try publishing the new version of current refactored environment. You will see some changes. And the "main" section of [package.json](https://docs.npmjs.com/files/package.json#main) will be wrong.
And yes - got the point what you talking about. But keep in mind that by using UMD is a good way to test for users your component for example in jsbin.com

**Comment by FDiskas** [comment]
@Sharlaan any news about this?

File: package.json [hunk 2]
```diff
@@ -70,6 +70,7 @@
     "react-dom": "^16.1.0",
     "react-router": "^4.2.0",
     "react-router-dom": "^4.2.2",
-    "react-to-typescript-definitions": "^0.22.0"
+    "react-to-typescript-definitions": "^0.22.0",
+    "storybook-addon-material-ui": "^0.8.2"
```
Comment (Sharlaan) [inline package.json]: This PR is about type definition, not storybook. You shouldnot mix.

Storybook (and htis addon) are already scheduled, so useless to add this here or any PR.

**Review by Sharlaan — CHANGES_REQUESTED** [review]
This PR is about type definition, not storybook. You shouldnot mix.

Storybook (and htis addon) are already scheduled, so useless to add this here or any PR.

File: package.json (hunk 2 above)
Comment (FDiskas) [reply package.json]: I just merged the master branch

Comment (FDiskas) [reply package.json]: You should check https://github.com/Sharlaan/material-ui-superselectfield/pull/94/files

Comment (Sharlaan) [reply package.json]: Ah ok, probably Max started on storybook.

**Comment by FDiskas** [comment]
DO something. Can't wait and fix merge conflicts to often

**Comment by KnisterPeter** [comment]
Added support for `prop-types`: https://github.com/KnisterPeter/react-to-typescript-definitions/issues/439#issuecomment-346266699

**Comment by FDiskas** [comment]
cool :jeans:

**Comment by FDiskas** [comment]
```
Conflicting files
package-lock.json
```
so why I'm not using npm any more. Use yarn instead :)

File: lib/SuperSelectField.d.ts (hunk 1 above)
Comment (FDiskas) [reply lib/SuperSelectField.d.ts]: Just info :)

**Comment by FDiskas** [comment]
> **Note** definitions should not be committed. They must be generated on each build.

I'm right?

**Pull request closed without merging**